### PR TITLE
Fix the Puppetmaster build and the Vagrant dev environment for Puppet

### PR DESCRIPTION
**Pull request by dougneal**
Fix the Puppetmaster build:
 * Drop the `govuk_puppetdb` module and merge its functionality into the `puppet` module
 * Drop the legacy Puppetmaster (Unicorn) entirely - all new builds to be on Ubuntu Trusty with PuppetServer

In order to test the build, I also needed to get Vagrant working when the `$::aws_migration` top-scope variable is set.

This caused tests to fail, which on investigation turned out to be because tests were running against the legacy `hieradata` directory. Added some logic to point them at `hieradata_aws` except in the case of legacy node classes that are not included in the AWS migration.

### CODE REVIEW

**Comment by issyl0** [comment]
Please can you test this on AWS staging?

**Comment by dougneal** [comment]
@issyl0 this is currently underway...

File: README.md [hunk 1]
```diff
@@ -185,51 +185,46 @@ runs the single test at line 7 of `collectd__package_spec.rb` with colour mode e
...
 Vagrant will run the Puppet provisioner against the node when it boots up.
 Nodes should look almost identical to that of our real
-production/staging/preview environments, including network addresses. To
-access a node's services like HTTP/HTTPS you can point your `hosts` file to
-the host-only IP address (eth1).
-
-Physical attributes like `memory` and `num_cores` will be ignored because
-they don't scale appropriately to local VMs, but can still be customised as
-described below.
+production/staging/preview environments.
```
Comment (issyl0) [inline README.md]: "Preview" was renamed to "Integration".

File: modules/puppet/manifests/init.pp [hunk 2]
```diff
@@ -12,18 +12,31 @@
...
   validate_bool($future_parser)
   validate_bool($use_puppetmaster)
 
-  include puppet::cronjob
-  include puppet::repository
-  include puppet::package
-  include puppet::monitoring
+  include '::puppet::cronjob'
+  contain '::puppet::cronjob'
+
```
Comment (issyl0) [inline modules/puppet/manifests/init.pp]: I'm not sure I understand the difference between `include` and `contain` - [this](https://puppet.com/docs/puppet/5.5/lang_containment.html#containing-classes) doesn't help, yet includes both examples. This is a more fundamental question than probably appropriate for this code review, though.

File: modules/puppet/spec/classes/puppet_spec.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@
     }}
 
     it do
-      is_expected.to contain_file('/etc/puppet/puppet.conf').with_content(/environmentpath = \/usr\/share\/puppet\/production\/current\/environments/)
+      is_expected.to contain_file('/etc/puppet/puppet.conf').with_content(/environmentpath = \/usr\/share\/puppet\/rp_env\/current\/environments/)
```
Comment (issyl0) [inline modules/puppet/spec/classes/puppet_spec.rb]: This looks like it's a variable - where does it come from? The `puppet.conf.erb` file contains `<%= scope.lookupvar('::environment') %>`.

File: vagrant.nodes.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+---
+defaults:
+  box: ubuntu/trusty64
+  memory: 3072
+
+nodes:
```
Comment (issyl0) [inline vagrant.nodes.yaml]: These are AWS node names (eg `db-admin`), not Carrenza names. Are we going the AWS way with the vagrant setup? 🚀

File: modules/puppet/manifests/init.pp (hunk 2 above)
Comment (dougneal) [reply modules/puppet/manifests/init.pp]: Containment means that if a resource forms an ordering relationship with the containing class, it also forms the same ordering relationship with the contained class.

From another look at the docs it looks like the `include` might be redundant in this case.. I'll try taking that out.

File: modules/puppet/spec/classes/puppet_spec.rb (hunk 3 above)
Comment (dougneal) [reply modules/puppet/spec/classes/puppet_spec.rb]: `rp_env` is what rspec-puppet sets `$::environment` to. I'll add a comment to clarify that.

File: vagrant.nodes.yaml (hunk 4 above)
Comment (dougneal) [reply vagrant.nodes.yaml]: That was my thinking, yes - make the Vagrant dev environment representative of the future...

**Comment by dougneal** [comment]
I'm going to close this PR and split it up into branches of a more manageable size, that will allow us to fix the Puppet Master build on AWS before tackling the Carrenza legacy.

**Pull request closed without merging**